packages/map-template/src/components/LegendDialog/LegendDialog.jsx

```diff
@@ -0,0 +1,93 @@
+import getLegendSectionsHeight from "../../helpers/GetLegendSectionsHeight";
+import getLegendSortedFields from "../../helpers/GetLegendSortedFields";
+
+/**
+ * Handle the Legend dialog.
+ *
+ */
+function LegendDialog() {
+    const { t } = useTranslation();
+    const primaryColorProp = useRecoilValue(primaryColorState);
+
+    const kioskLocation = useRecoilValue(kioskLocationState)
+
+    const legendSections = getLegendSortedFields(kioskLocation);
+
+    const isKioskContext = useIsKioskContext();
+
+    const legendModalRef = useRef();
+
+    const legendSectionRef = useRef();
+
+    const scrollButtonsRef = useRef();
+
+    const [showLegendDialog, setShowLegendDialog] = useRecoilState(showLegendDialogState);
+
+    const [legendSize, setLegendSize] = useRecoilState(legendSizeState);
+
+    const [showScrollButtons, setShowScrollButtons] = useState(false);
+
+    /*
+     * Get the height of the legend sections. 
+     * Set the legend size atom and determine 
+     * if the scroll buttons should be shown.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext) {
+            getLegendSectionsHeight().then(padding => {
+                if (padding > 700) {
+                    setLegendSize(padding);
+                    setShowScrollButtons(true);
+                } else {
+                    setShowScrollButtons(false);
+                }
+            });
+        }
+    }, [showLegendDialog, isKioskContext]);
+
+    /*
+     * Setup scroll buttons to scroll in legend sections when in kiosk mode.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext && showScrollButtons) {
+            const legendContent = document.querySelector('.legend__sections');
+            scrollButtonsRef.current.scrollContainerElementRef = legendContent;
+        }
+    }, [showLegendDialog, legendSections, showScrollButtons]);
+
+    return (<>
+        <div className="background"></div>
+        <div className="legend" ref={legendModalRef}>
+            {legendSections.length > 0 &&
+                <div className={`legend__sections ${legendSize > 700 ? 'legend__sections--scrollable' : ''}`}
+                    style={{ maxHeight: legendSize > 700 ? '700px' : 'auto' }}>
+                    {legendSections.map((legendSection, index) => <div key={index} className="legend__section" ref={legendSectionRef}>
+                        <div className="legend__heading">{legendSection.heading}</div>
+                        <div className="legend__content">{legendSection.content}</div>
+                    </div>)}
+                </div>}
+            <button className="legend__button" style={{ background: primaryColorProp }} onClick={() => setShowLegendDialog(false)}>{t('Close')}</button>
+        </div>
+
+        { /* Buttons to scroll in the list of search results if in kiosk context */}
+        {isKioskContext && showLegendDialog && showScrollButtons && createPortal(
```

Comment:
Is the createPortal necessary when it's placed within the same React component?

And if it can be removed, will that solve the issue that we introduced the MutationObserver for?

Reply:
I think the createPortal and the Mutation observer are not necessary anymore. Refactored the code and pushed the changes.

Reply:
🥳